doc/functions.dx

```diff
@@ -242,23 +242,16 @@ them in parens like `def`.
 map (\(a:Float). a + 1) [1., 2., 3.]
 > [2., 3., 4.]
 
-'Like `def`, we can also specify implicit parameters with curly braces,
+'Also like `def`, we can specify implicit parameters with curly braces:
 
 \{a} (x:a). x
 > \a:Type ?-> \x:a. x
 
-'whose types are inferred if not annotated.  However, the `\`
-syntax does not allow listing interface constraints, because that
-conflicts with pattern-matching a table argument (see [Issue
-954](https://github.com/google-research/dex-lang/issues/954)).
-The (unfortunate) error message looks like this, should you try it:
+'and class parameters with square brackets:
 
 \{n} [Ix n] (xs:n=>Float). sum xs
-> Type error:Not a data constructor: Ix
->
-> \{n} [Ix n] (xs:n=>Float). sum xs
->       ^^^^
-
-'The workaround is to either use a `def`, or to annotate the type of
-the anonymous function.  If the type include interface constraints,
-the compiler will insert an appropriate parameter to accept them.
+> \n:Type ?->
+>   \v#0:(Ix n) ?=>
+>     \xs:(n => Float32).
+>       v#1:Float32 = sum n Float32 (Instance Add.2 []) v#0 xs
+>       v#1
```

Comment:
Perhaps print the type of this function instead of its implementation?

Reply:
Done.